iwyu_output.cc

```diff
@@ -613,6 +627,28 @@ void IwyuFileInfo::ReportForwardDeclareUse(SourceLocation use_loc,
   LogSymbolUse("Marked fwd-decl use of decl", symbol_uses_.back());
 }
 
+void IwyuFileInfo::ReportUsingDeclUse(SourceLocation use_loc,
+                                      const UsingDecl* using_decl,
+                                      bool in_cxx_method_body,
+                                      const char* comment) {  
+  // If accessing a symbol through a using decl in the same file that contains
+  // the using decl, we must mark the using decl as referenced. At the end of
+  // traversing the AST, we check to see if a using decl is unreferenced and
+  // add a full use of one of its shadow decls so that the source file
+  // continues to compile.
+  map<const UsingDecl*, bool>::iterator using_decl_status = 
+    using_decl_referenced_.find(using_decl);
+
+  if (using_decl_status != using_decl_referenced_.end()) {
+    using_decl_status->second = true;
+  }
+
```

Comment:
Oh, thanks for the explanation. So `using_decl_referenced_` is really `using_decl_referenced_in_same_file_`? I'm not suggesting a name change (too painfully long), but it might be worth making that clearer somehow.